astropy/modeling/spline.py

```diff
@@ -0,0 +1,438 @@
+                setattr(self, attribute, None)
+            else:
+                # No options passed in and No options set
+                optional_inputs[arg] = self.optional_inputs[arg]
+
+        return optional_inputs
+
+
+class Spline1D(Fittable1DModel, _Spline):
+    """
+    One dimensional Spline model
+
+    Parameters
+    ----------
+    k : int, optional
+        Degree of the smoothing spline. Must be 1 <= k <= 5. Default is
+        k = 3, a cubic spline.
+    ext : int or str, optional
+        Controls the extrapolation mode for elements not in the interval
+        defined by the knot sequence.
+
+        if ext=0 or ‘extrapolate’, return the extrapolated value.
+
+        if ext=1 or ‘zeros’, return 0
+
+        if ext=2 or ‘raise’, raise a ValueError
+
+        if ext=3 of ‘const’, return the boundary value.
+
+        The default value is 0.
+
+    check_finite : bool, optional
+        Whether to check that the input arrays contain only finite
+        numbers. Disabling may give a performance gain, but may result
+        in problems (crashes, non-termination or non-sensical results)
+        if the inputs do contain infinities or NaNs. Default is False.
+
+    Notes
+    -----
+    This is largely a wrapper of the `scipy.interpolate.UnivariateSpline`
+    class and its child `scipy.interpolate.LSQUnivariateSpline`. See
+    scipy.interpolate for more details.
+    """
+
+    optional_inputs = {'nu': 0}
+
+    def __init__(self, k=3, ext=0, check_finite=False, n_models=None,
+                 model_set_axis=None, name=None, meta=None, **params):
+        self._k = k
+        self._ext = ext
+        self._check_finite = check_finite
+        self._spline = None
+        super().__init__(n_models=n_models, model_set_axis=model_set_axis,
+                         name=name, meta=meta, **params)
+
+        # Hack to allow an optional model argument
+        self._create_optional_inputs()
+
+    def evaluate(self, x, **kwargs):
+        """
+        Evaluate the model
+
+        Parameters
+        ----------
+        x : array_like
+            A 1-D array of points at which to return the value of the smoothed
+            spline or its derivatives. Note: `x` can be unordered but the
+            evaluation is more efficient if `x` is (partially) ordered.
+        nu : int
+            The order of derivative of the spline to compute.
+        """
+
+        # Hack to allow an optional model argument
+        kwargs = self._get_optional_inputs(**kwargs)
+
+        return self.spline(x, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+
+        # Hack to allow an optional model argument
+        kwargs = self._intercept_optional_inputs(**kwargs)
+
+        return super().__call__(*args, **kwargs)
+
+    def fit_spline(self, x, y, w=None, s=None):
+        """
+        Fit spline using `scipy.interpolate.UnivariateSpline`
+
+        Parameters
+        ----------
```

Comment:
I think there should be an option where the user selects a fixed number of knots (or eventually, a list of the knot locations) rather than relying on a fitting threshold.

Reply:
This is performed by `scipy.interpolate.LSQUnivartateSpline` wrapper `fit_LSQ_spline`.